[Bug] Internal Server Error with Custom Cover Image

**RomM version**

```
# docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
rommapp/romm        latest              c1f46b652d32        8 days ago          289MB
mariadb             latest              2af45915db5d        10 days ago         414MB
```

**Describe the bug**

Image is not accepted by romm

**To Reproduce**

Steps to reproduce the behavior:

I wanted to use this image: http://adb.arcadeitalia.net/media/mame.current/titles/ldrun.png?release=208 as custom cover.

**Expected behavior**

I got an empty error message from romm and below logs

**Screenshots**

```text
romm       | 192.168.192.20:0 - "PUT /api/roms/3?rename_as_source=false&remove_cover=false&unmatch_metadata=false HTTP/1.0" 500
romm       | INFO:	  [nginx][2024-09-13 12:57:40]	192.168.192.20 - - "PUT /api/roms/3?rename_as_source=false&remove_cover=false&unmatch_metadata=false HTTP/1.1" 500 21 "http://kodiwyse.local:6464/rom/3" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36" rt=0.128 uct="0.000" uht="0.127" urt="0.127"
romm       | [2024-09-13 12:57:40 +0000] [23] [ERROR] Exception in ASGI application
romm       |   + Exception Group Traceback (most recent call last):
romm       |   |   File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
romm       |   |     yield
romm       |   |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 190, in __call__
romm       |   |     async with anyio.create_task_group() as task_group:
romm       |   |   File "/src/.venv/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 680, in __aexit__
romm       |   |     raise BaseExceptionGroup(
romm       |   | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
romm       |   +-+---------------- 1 ----------------
romm       |     | Traceback (most recent call last):
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
romm       |     |     result = await app(  # type: ignore[func-returns-value]
romm       |     |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
romm       |     |     return await self.app(scope, receive, send)
romm       |     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
romm       |     |     await super().__call__(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 123, in __call__
romm       |     |     await self.middleware_stack(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 186, in __call__
romm       |     |     raise exc
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 164, in __call__
romm       |     |     await self.app(scope, receive, _send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 189, in __call__
romm       |     |     with collapse_excgroups():
romm       |     |   File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
romm       |     |     self.gen.throw(value)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
romm       |     |     raise exc
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 191, in __call__
romm       |     |     response = await self.dispatch_func(request, call_next)
romm       |     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/backend/utils/context.py", line 41, in set_context_middleware
romm       |     |     return await call_next(request)
romm       |     |            ^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 165, in call_next
romm       |     |     raise app_exc
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 151, in coro
romm       |     |     await self.app(scope, receive_or_disconnect, send_no_error)
romm       |     |   File "/backend/handler/auth/middleware.py", line 146, in __call__
romm       |     |     await self.app(scope, receive, send_wrapper)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
romm       |     |     await self.app(scope, receive, send)
romm       |     |   File "/backend/handler/auth/middleware.py", line 19, in __call__
romm       |     |     await super().__call__(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
romm       |     |     await self.app(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 91, in __call__
romm       |     |     await self.simple_response(scope, receive, send, request_headers=headers)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 146, in simple_response
romm       |     |     await self.app(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
romm       |     |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
romm       |     |     raise exc
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
romm       |     |     await app(scope, receive, sender)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 758, in __call__
romm       |     |     await self.middleware_stack(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 778, in app
romm       |     |     await route.handle(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 299, in handle
romm       |     |     await self.app(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 79, in app
romm       |     |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
romm       |     |     raise exc
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
romm       |     |     await app(scope, receive, sender)
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 74, in app
romm       |     |     response = await func(request)
romm       |     |                ^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 278, in app
romm       |     |     raw_response = await run_endpoint_function(
romm       |     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
romm       |     |     return await dependant.call(**values)
romm       |     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/starlette/authentication.py", line 84, in async_wrapper
romm       |     |     return await func(*args, **kwargs)
romm       |     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/backend/endpoints/rom.py", line 453, in update_rom
romm       |     |     fs_resource_handler.resize_cover_to_small(file_location_s)
romm       |     |   File "/backend/handler/filesystem/resources_handler.py", line 38, in resize_cover_to_small
romm       |     |     cover = Image.open(cover_path)
romm       |     |             ^^^^^^^^^^^^^^^^^^^^^^
romm       |     |   File "/src/.venv/lib/python3.12/site-packages/PIL/Image.py", line 3498, in open
romm       |     |     raise UnidentifiedImageError(msg)
romm       |     | PIL.UnidentifiedImageError: cannot identify image file '/romm/resources/roms/14/3/cover/small.png'
romm       |     +------------------------------------
romm       | During handling of the above exception, another exception occurred:
romm       | Traceback (most recent call last):
romm       |   File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
romm       |     result = await app(  # type: ignore[func-returns-value]
romm       |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
romm       |     return await self.app(scope, receive, send)
romm       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
romm       |     await super().__call__(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 123, in __call__
romm       |     await self.middleware_stack(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 186, in __call__
romm       |     raise exc
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 164, in __call__
romm       |     await self.app(scope, receive, _send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 189, in __call__
romm       |     with collapse_excgroups():
romm       |   File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
romm       |     self.gen.throw(value)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
romm       |     raise exc
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 191, in __call__
romm       |     response = await self.dispatch_func(request, call_next)
romm       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/backend/utils/context.py", line 41, in set_context_middleware
romm       |     return await call_next(request)
romm       |            ^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 165, in call_next
romm       |     raise app_exc
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 151, in coro
romm       |     await self.app(scope, receive_or_disconnect, send_no_error)
romm       |   File "/backend/handler/auth/middleware.py", line 146, in __call__
romm       |     await self.app(scope, receive, send_wrapper)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
romm       |     await self.app(scope, receive, send)
romm       |   File "/backend/handler/auth/middleware.py", line 19, in __call__
romm       |     await super().__call__(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
romm       |     await self.app(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 91, in __call__
romm       |     await self.simple_response(scope, receive, send, request_headers=headers)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 146, in simple_response
romm       |     await self.app(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
romm       |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
romm       |     raise exc
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
romm       |     await app(scope, receive, sender)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 758, in __call__
romm       |     await self.middleware_stack(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 778, in app
romm       |     await route.handle(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 299, in handle
romm       |     await self.app(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 79, in app
romm       |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
romm       |     raise exc
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
romm       |     await app(scope, receive, sender)
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 74, in app
romm       |     response = await func(request)
romm       |                ^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 278, in app
romm       |     raw_response = await run_endpoint_function(
romm       |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
romm       |     return await dependant.call(**values)
romm       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/starlette/authentication.py", line 84, in async_wrapper
romm       |     return await func(*args, **kwargs)
romm       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/backend/endpoints/rom.py", line 453, in update_rom
romm       |     fs_resource_handler.resize_cover_to_small(file_location_s)
romm       |   File "/backend/handler/filesystem/resources_handler.py", line 38, in resize_cover_to_small
romm       |     cover = Image.open(cover_path)
romm       |             ^^^^^^^^^^^^^^^^^^^^^^
romm       |   File "/src/.venv/lib/python3.12/site-packages/PIL/Image.py", line 3498, in open
romm       |     raise UnidentifiedImageError(msg)
romm       | PIL.UnidentifiedImageError: cannot identify image file '/romm/resources/roms/14/3/cover/small.png'
```

**Desktop (please complete the following information):**

- OS: Mac OS
- Browser Vivaldi
- Version 6.9.3447.37 (Stable channel) (x86_64)

**Smartphone (please complete the following information):**

not used

--- a/backend/endpoints/collections.py
+++ b/backend/endpoints/collections.py
@@ -1,7 +1,8 @@
 import json
+from io import BytesIO
 from shutil import rmtree
 
-from anyio import open_file
+from anyio import Path
 from config import RESOURCES_BASE_PATH
 from decorators.auth import protected_route
 from endpoints.responses import MessageResponse
@@ -17,6 +18,7 @@ from handler.filesystem import fs_resource_handler
 from handler.filesystem.base_handler import CoverSize
 from logger.logger import log
 from models.collection import Collection
+from PIL import Image
 from sqlalchemy.inspection import inspect
 from utils.router import APIRouter
 
@@ -62,15 +64,14 @@ async def add_collection(
             artwork_path,
         ) = await fs_resource_handler.build_artwork_path(_added_collection, file_ext)
 
-        artwork_file = artwork.file.read()
-        file_location_s = f"{artwork_path}/small.{file_ext}"
-        async with await open_file(file_location_s, "wb+") as artwork_s:
-            await artwork_s.write(artwork_file)
-            fs_resource_handler.resize_cover_to_small(file_location_s)
-
-        file_location_l = f"{artwork_path}/big.{file_ext}"
-        async with await open_file(file_location_l, "wb+") as artwork_l:
-            await artwork_l.write(artwork_file)
+        artwork_content = BytesIO(await artwork.read())
+        file_location_small = Path(f"{artwork_path}/small.{file_ext}")
+        file_location_large = Path(f"{artwork_path}/big.{file_ext}")
+        with Image.open(artwork_content) as img:
+            img.save(file_location_large)
+            fs_resource_handler.resize_cover_to_small(
+                img, save_path=file_location_small
+            )
     else:
         path_cover_s, path_cover_l = await fs_resource_handler.get_cover(
             overwrite=True,
@@ -183,15 +184,15 @@ async def update_collection(
             cleaned_data["path_cover_l"] = path_cover_l
             cleaned_data["path_cover_s"] = path_cover_s
 
-            artwork_file = artwork.file.read()
-            file_location_s = f"{artwork_path}/small.{file_ext}"
-            async with await open_file(file_location_s, "wb+") as artwork_s:
-                await artwork_s.write(artwork_file)
-                fs_resource_handler.resize_cover_to_small(file_location_s)
+            artwork_content = BytesIO(await artwork.read())
+            file_location_small = Path(f"{artwork_path}/small.{file_ext}")
+            file_location_large = Path(f"{artwork_path}/big.{file_ext}")
+            with Image.open(artwork_content) as img:
+                img.save(file_location_large)
+                fs_resource_handler.resize_cover_to_small(
+                    img, save_path=file_location_small
+                )
 
-            file_location_l = f"{artwork_path}/big.{file_ext}"
-            async with await open_file(file_location_l, "wb+") as artwork_l:
-                await artwork_l.write(artwork_file)
             cleaned_data.update({"url_cover": ""})
         else:
             if data.get("url_cover", "") != collection.url_cover or not (

--- a/backend/endpoints/rom.py
+++ b/backend/endpoints/rom.py
@@ -1,10 +1,11 @@
 import binascii
 from base64 import b64encode
+from io import BytesIO
 from shutil import rmtree
 from typing import Annotated
 from urllib.parse import quote
 
-from anyio import Path, open_file
+from anyio import Path
 from config import (
     DEV_MODE,
     DISABLE_DOWNLOAD_ENDPOINT_AUTH,
@@ -23,6 +24,7 @@ from handler.filesystem import fs_resource_handler, fs_rom_handler
 from handler.filesystem.base_handler import CoverSize
 from handler.metadata import meta_igdb_handler, meta_moby_handler
 from logger.logger import log
+from PIL import Image
 from starlette.requests import ClientDisconnect
 from starlette.responses import FileResponse
 from streaming_form_data import StreamingFormDataParser
@@ -433,15 +435,15 @@ async def update_rom(
                 {"path_cover_s": path_cover_s, "path_cover_l": path_cover_l}
             )
 
-            artwork_file = artwork.file.read()
-            file_location_s = f"{artwork_path}/small.{file_ext}"
-            async with await open_file(file_location_s, "wb+") as artwork_s:
-                await artwork_s.write(artwork_file)
-                fs_resource_handler.resize_cover_to_small(file_location_s)
+            artwork_content = BytesIO(await artwork.read())
+            file_location_small = Path(f"{artwork_path}/small.{file_ext}")
+            file_location_large = Path(f"{artwork_path}/big.{file_ext}")
+            with Image.open(artwork_content) as img:
+                img.save(file_location_large)
+                fs_resource_handler.resize_cover_to_small(
+                    img, save_path=file_location_small
+                )
 
-            file_location_l = f"{artwork_path}/big.{file_ext}"
-            async with await open_file(file_location_l, "wb+") as artwork_l:
-                await artwork_l.write(artwork_file)
             cleaned_data.update({"url_cover": ""})
         else:
             if data.get("url_cover", "") != rom.url_cover or not (

--- a/backend/handler/filesystem/resources_handler.py
+++ b/backend/handler/filesystem/resources_handler.py
@@ -7,7 +7,7 @@ from fastapi import HTTPException, status
 from logger.logger import log
 from models.collection import Collection
 from models.rom import Rom
-from PIL import Image
+from PIL import Image, ImageFile
 from utils.context import ctx_httpx_client
 
 from .base_handler import CoverSize, FSHandler
@@ -33,9 +33,8 @@ class FSResourcesHandler(FSHandler):
         return False
 
     @staticmethod
-    def resize_cover_to_small(cover_path: str) -> None:
-        """Path of the cover image to resize"""
-        cover = Image.open(cover_path)
+    def resize_cover_to_small(cover: ImageFile.ImageFile, save_path: Path) -> None:
+        """Resize cover to small size, and save it to filesystem."""
         if cover.height >= 1000:
             ratio = 0.2
         else:
@@ -44,7 +43,7 @@ class FSResourcesHandler(FSHandler):
         small_height = int(cover.height * ratio)
         small_size = (small_width, small_height)
         small_img = cover.resize(small_size)
-        small_img.save(cover_path)
+        small_img.save(save_path)
 
     async def _store_cover(
         self, entity: Rom | Collection, url_cover: str, size: CoverSize
@@ -57,15 +56,15 @@ class FSResourcesHandler(FSHandler):
             url_cover: url to get the cover
             size: size of the cover
         """
-        cover_file = f"{size.value}.png"
-        cover_path = f"{RESOURCES_BASE_PATH}/{entity.fs_resources_path}/cover"
+        cover_path = Path(f"{RESOURCES_BASE_PATH}/{entity.fs_resources_path}/cover")
+        cover_file = cover_path / Path(f"{size.value}.png")
 
         httpx_client = ctx_httpx_client.get()
         try:
             async with httpx_client.stream("GET", url_cover, timeout=120) as response:
                 if response.status_code == 200:
-                    await Path(cover_path).mkdir(parents=True, exist_ok=True)
-                    async with await open_file(f"{cover_path}/{cover_file}", "wb") as f:
+                    await cover_path.mkdir(parents=True, exist_ok=True)
+                    async with await cover_file.open("wb") as f:
                         async for chunk in response.aiter_raw():
                             await f.write(chunk)
         except httpx.NetworkError as exc:
@@ -77,7 +76,8 @@ class FSResourcesHandler(FSHandler):
             log.warning(f"Failure writing cover {url_cover} to file (ProtocolError)")
 
         if size == CoverSize.SMALL:
-            self.resize_cover_to_small(f"{cover_path}/{cover_file}")
+            with Image.open(cover_file) as img:
+                self.resize_cover_to_small(img, save_path=cover_file)
 
     @staticmethod
     async def _get_cover_path(entity: Rom | Collection, size: CoverSize) -> str:

